components/director/internal/domain/apptemplate/resolver.go

```diff
@@ -745,28 +747,41 @@ func validateAppTemplateNameBasedOnProvider(name string, appInput *graphql.Appli
 }
 
 func (r *Resolver) checkProviderAppTemplateExistence(ctx context.Context, labels map[string]interface{}) error {
-	distinguishLabelKey := r.selfRegManager.GetSelfRegDistinguishingLabelKey()
+	selfRegisterDistinguishLabelKey := r.selfRegManager.GetSelfRegDistinguishingLabelKey()
 	regionLabelKey := selfregmanager.RegionLabel
-
-	distinguishLabelValue, distinguishLabelExists := labels[distinguishLabelKey]
 	region, regionExists := labels[regionLabelKey]
 
-	if distinguishLabelExists && regionExists {
+	distinguishLabelKeys := []string{selfRegisterDistinguishLabelKey, r.appTemplateProductLabel}
+	appTemplateDistinguishLabels := make(map[string]interface{}, len(distinguishLabelKeys))
+
+	for _, key := range distinguishLabelKeys {
+		if value, exists := labels[key]; exists {
+			appTemplateDistinguishLabels[key] = value
+		}
+	}
+
+	for labelKey, labelValue := range appTemplateDistinguishLabels {
+		msg := fmt.Sprintf("%q: %q", labelKey, labelValue)
+
 		filters := []*labelfilter.LabelFilter{
-			labelfilter.NewForKeyWithQuery(distinguishLabelKey, fmt.Sprintf("\"%s\"", distinguishLabelValue)),
-			labelfilter.NewForKeyWithQuery(regionLabelKey, fmt.Sprintf("\"%s\"", region)),
+			labelfilter.NewForKeyWithQuery(labelKey, fmt.Sprintf("\"%s\"", labelValue)),
+		}
+
+		if regionExists {
+			filters = append(filters, labelfilter.NewForKeyWithQuery(regionLabelKey, fmt.Sprintf("\"%s\"", region)))
+			msg += fmt.Sprintf(" and %q: %q", regionLabelKey, region)
 		}
 
-		log.C(ctx).Infof("Getting application template for labels %q: %q and %q: %q", regionLabelKey, region, distinguishLabelKey, distinguishLabelValue)
+		log.C(ctx).Infof("Getting application template for labels %s", msg)
 		appTemplate, err := r.appTemplateSvc.GetByFilters(ctx, filters)
```

Comment:
If the filters are applied with an "AND" in the query will it even be possible to find an app template which has all these 2 or 3 labels? Can we check that?

Reply:
Of course. There currently we are looking for a maximum of 2 labels at once - region + self reg OR region + cld. Why wouldn't we find App Templates by this criteria?

Reply:
when i double check it - all is good